cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -477,17 +470,122 @@ func formatPnpmKeyV6(name string, version string) string {
 }
 
 func getVersionFromKey(key string) string {
-	atIndex := strings.LastIndex(key, "/")
-	if atIndex == -1 || len(key) == atIndex+1 {
-		return ""
+	dp := parseDepPath(key)
+	if dp.peerSuffix != "" {
+		return fmt.Sprintf("%s_%s", dp.version, dp.peerSuffix)
 	}
-	return key[atIndex+1:]
+	return dp.version
```

Comment:
What about moving these into `PnpmLockfile` as methods?

It already has `.isV6`, so you could switch there, or alternatively a `PnpmLockfile` interface with `DecodePnpmLockfile` returning a `PnpmLockfile5` or a `PnpmLockfile6`?

(`PnpmLockfile` would probably end up renamed to something like `RawPnpmLockfile`.)